src/libsync/syncengine.cpp

```diff
@@ -1067,7 +1068,11 @@ void SyncEngine::slotDiscoveryJobFinished(int discoveryResult)
     setNetworkLimits(_uploadLimit, _downloadLimit);
 
     // enable/disable delta sync
-    setDeltaSync(_deltaSyncEnabled, _deltaSyncMinFileSize);
+    ConfigFile cfgFile;
+    if (_deltaSyncEnabled)
+        setDeltaSync(_deltaSyncEnabled, _deltaSyncMinFileSize);
+    else
+        setDeltaSync(cfgFile.deltaSyncEnabled(), cfgFile.deltaSyncMinFileSize());
```

Comment:
I know I suggested this, but I think I didn't think this through. Your two new sync options should actually be inside the `SyncOptions` struct. That would simplify the code since it's already available in SyncEngine and OwncloudPropagator.

Also, the ConfigFile is something that belongs into the UI part of the code and shouldn't be used in SyncEngine. For the other options it's the `Folder` object that prepares the `SyncOptions` hands it to the SyncEngine. And cmd's command line processing does the same.

Reply:
ok, no problem will look into `SyncOptions`, yeah when I did the above it felt clunky/wrong :)

Reply:
Done.